RecoTracker/MkFitCore/src/CandCloner.h

```diff
@@ -0,0 +1,169 @@
+#ifndef RecoTracker_MkFitCore_src_CandCloner_h
+#define RecoTracker_MkFitCore_src_CandCloner_h
+
+#include "MkFinder.h"
+
+#include <vector>
+
+namespace mkfit {
+
+  class IterationParams;
+  class EventOfCombCandidates;
+
+  //#define CC_TIME_LAYER
+  //#define CC_TIME_ETA
+
+  typedef std::pair<int, int> CandClonerWork_t;
+
+  class CandCloner {
+  public:
+    // Maximum number of seeds processed in one call to ProcessSeedRange()
+    static const int s_max_seed_range = MPT_SIZE;
+
+  private:
+    // Temporaries in ProcessSeedRange(), resized/reserved  in constructor.
+
+    // Size of this one is s_max_seed_range
+    std::vector<std::vector<TrackCand>> t_cands_for_next_lay;
+
+  public:
+    CandCloner() { t_cands_for_next_lay.resize(s_max_seed_range); }
+
+    void Setup(const IterationParams &ip);
+    void Release();
+
+    ~CandCloner() {}
+
+    void begin_eta_bin(EventOfCombCandidates *e_o_ccs,
+                       std::vector<std::pair<int, int>> *update_list,
+                       std::vector<std::vector<TrackCand>> *extra_cands,
+                       int start_seed,
+                       int n_seeds) {
+      mp_event_of_comb_candidates = e_o_ccs;
+      mp_kalman_update_list = update_list;
+      mp_extra_cands = extra_cands;
+      m_start_seed = start_seed;
+      m_n_seeds = n_seeds;
+      m_hits_to_add.resize(n_seeds);
+
+      // XXX Should resize vectors in m_hits_to_add to whatever makes sense ???
+      // for (int i = 0; i < n_seeds; ++i)
+      // {
+      //   m_hits_to_add[i].reserve(20 * Config::maxCandsPerSeed);
+      // }
```

Comment:
remove

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/842cdaf057f74ecf568f12bb1a4cb96e19c5d22c